cupyx/fallback_mode/fallback.py

```diff
@@ -36,14 +55,15 @@ def __getattr__(self, attr):
             (_RecursiveAttr object, NumPy scalar):
             Returns_RecursiveAttr object with new numpy_object, cupy_object.
             Returns scalars if requested.
+            Returns objects in cupy which is an alias of numpy object.
         """
 
         # getting attr
         numpy_object = getattr(self._numpy_object, attr)
         cupy_object = getattr(self._cupy_object, attr, None)
 
-        # Retrieval of NumPy scalars
-        if isinstance(numpy_object, np.ScalarType):
+        # if same objects, then return
+        if numpy_object is cupy_object:
             return numpy_object
 
```

Comment:
How about adding here the following code to fix https://github.com/cupy/cupy/pull/2272/files#r304249518?
```py
if numpy_object is np.ndarray:
    return ndarray
```

Reply:
https://github.com/cupy/cupy/pull/2272#discussion_r304247271
https://github.com/cupy/cupy/pull/2272#discussion_r304249518
Both of the above bugs arises due to `type(numpy.array([1]))` resulting in `fallback.array`.
I didn't find any simple way to override `type` method in Python. One solution would be to force user to use `__class__` instead of `type`, as `__class__` could be overridden as follows in `class ndarray`:
```python
@property
def __class__(self):
    return numpy.ndarray # numpy -> fallback.numpy
```
As for this comment https://github.com/cupy/cupy/pull/2272/files#r304249518, we've `isinstance` overridden. Because the solution you proposed won't work for below code:
```python
from cupyx.fallback_mode import numpy
numpy.ndarray((2, 3), 'f')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes 2 positional arguments but 3 were given
```

Reply:
Even if we don't resolve above bugs, I think that following will be a nice addition:
> ```python
>    @property
>    def __class__(self):
>       return numpy.ndarray # numpy -> fallback.numpy
> ```
which would result in following:
```python
from cupyx.fallback_mode import numpy
numpy.array([1, 2, 3]).__class__
<numpy = <class 'numpy.ndarray'>, cupy = <class 'cupy.core.core.ndarray'>>
```

Reply:
Are there some problems with my suggestion: https://github.com/cupy/cupy/pull/2272#discussion_r304249866 ?

Reply:
Yes, it'll result in following due to `numpy.ndarray` returning `fallback.ndarray` which does not have `__init__` similar to `np.ndarray`:
```python
from cupyx.fallback_mode import numpy
numpy.ndarray((2, 3), 'f')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes 2 positional arguments but 3 were given
````

Reply:
So you should fix `ndarray.__init__` (and `_get_fallback_result`) to avoid it.

Reply:
I've added `ndarray.__new__` and made some modifications so that, ndarray can be initialized with `cupy.ndarray` as well as support call to `cupy.ndarray.__init__`.